### DISCUSSION TITLE
@import directive not resolved by postcss-import

### DISCUSSION THREAD

**Discussion by revosw** [tailwindlabs/tailwindcss#3173] [Help] [answered]
[Here is the project relating to the problem](https://github.com/revosw/learnwiremod)

There are three [source css files](https://github.com/revosw/learnwiremod/tree/master/src) `highlight`, `scrollbar` and `styles` being imported along with tailwind inside `consolidate`. However, postcss-import can't resolve my own css files, only the tailwind css files. Any ideas?

This image shows that postcss-import failed to resolve my css files
![](https://lbda.net/kIwh.png)

**Comment by adamwathan** [comment]
Hey! postcss-import does not allow you to have imports anywhere except at the very top of your file:

> This plugin attempts to follow the CSS @import spec; @import statements must precede all other statements (besides @charset).

https://github.com/postcss/postcss-import

**Reply by revosw** [reply]
I just want to make sure we both have the same mental image - when I run `npm run build:css`, it takes `consolidate.css` as input. That css file looks like this:
![](https://lbda.net/VwPz.png)

My build scripts look like this:
```
"scripts": {
    "build:css": "cross-env NODE_ENV=production tailwindcss build src/consolidate.css -o public/css/styles.css",
    "build:css-dev": "tailwindcss build src/consolidate.css -o public/css/styles.css"
  },
```

[Here is the output](https://hastebin.com/fujojebame.css) of the build, lines 687 - 691 has the unresolved imports

**Comment by adamwathan** [comment, marked as answer]
Ah the real problem here is you are using the `tailwindcss` CLI tool, that doesn't use postcss-import at all. So those imports are just not going to be processed period. You'll want to use `postcss-cli` instead and set up a postcss.config.js file if you want to use other PostCSS plugins.

**Reply by revosw** [reply]
Ah, thanks! Now everything clicked into place

**Comment by revosw** [comment]
### How I solved my problem (tailwind v2)
Do not use `tailwind build <source.css> -o <dest.css>`. Instead use `postcss <source.css> -o <dest.css>`

You need the dependencies **postcss**, **postcss-cli**, **postcss-import** and **tailwindcss**
`npm i -D postcss postcss-cli postcss-import tailwindcss`

My `postcss.config.js` looks about the same as how it's shown [in the documentation](https://tailwindcss.com/docs/using-with-preprocessors#build-time-imports):
```js
// postcss.config.js
module.exports = {
  plugins: [
    require('postcss-import'),
    require('tailwindcss')
  ]
}
```
My build scripts now look like this:
```json
"scripts": {
  "build:css": "postcss src/consolidate.css -o public/css/styles.css --env production",
  "build:css-dev": "postcss src/consolidate.css -o public/css/styles.css"
},
```
I have all the stylesheets inside `src/`, and one of the stylesheets consolidate all other stylesheets
```css
@import "tailwindcss/base";
@import "tailwindcss/components";
@import "tailwindcss/utilities";
@import "./highlight.css";
@import "./scrollbar.css";
@import "./styles.css"
```

**Reply by hungtienvu** [reply]
@revosw was your build:css minifying the css also? I've to use cssnano for the minifying.

**Reply by revosw** [reply]
I didn't do any minifying back then. However, now that I'm on tailwind 3, I have removed postcss as a dependency and my scripts look like this:
```json
"scripts": {
  "build:css": "tailwindcss -i src/consolidate.css -o public/css/styles.css --minify --no-autoprefixer",
  "dev:css": "tailwindcss -i src/consolidate.css -o public/css/styles.css --watch --no-autoprefixer"
},
```

**Reply by biki** [reply]
How are you resolving the imports to your css files without postcss? Thanks in advance

**Reply by revosw** [reply]
My project is so small that I chose to only have a single css file with all my custom styles, so I don't have to resolve imports
